src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/dynamic_coefficient.jl

```diff
@@ -2,13 +2,20 @@ using Oceananigans.Architectures: architecture
 using Oceananigans.Fields: interpolate
 using Statistics
 
-struct DynamicCoefficient{A, FT, S}
+mutable struct DynamicCoefficient{A, FT, S}
```

Comment:
`CPUDynamicCoefficient` for consistency? Then we can define `DynamicCoefficient = Union{CPUDynamicCoefficient, GPUDynamicCoefficient}` and we (I think) retain the same behavior for previously-defined methods methods.

Reply:
No because `DynamicCoefficient` is also used for GPU. The distinction between these two objects is more like `CuArray` versus `CuDeviceArray`. Both are for GPU, but only `CuDeviceArray` is used _inside_ kernels. Does that make sense? `GPUDynamicCoefficient` is not user facing, it only crops up internally inside kernels.

Reply:
made that change to clarify (there is also a `GPUFieldTimeSeries`, we can clean that up in another PR)